### PR TITLE
Finalise data frames containing unspecified columns

### LINKED ISSUE #205: Can't cast <logical> to <vctrs_unspecified>
``` r
library(vctrs)
df <- data.frame(y = NA, x = c(1, 2, NA))
vec_cast_common(df, df)
#> Error: Can't cast <logical> to <vctrs_unspecified>
```

<sup>Created on 2019-02-26 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1.9000)</sup>

### PR DESCRIPTION
**Pull request by lionel-**
Fixes #205

### CODE REVIEW

File: tests/testthat/test-unspecified.R [hunk 1]
```diff
@@ -34,3 +34,21 @@ test_that("subsetting works", {
 test_that("has useful print method", {
   expect_known_output(unspecified(), print = TRUE, file = test_path("test-unspecified.txt"))
 })
+
+test_that("can finalise data frame containing unspecified columns", {
+  df <- data.frame(y = NA, x = c(1, 2, NA))
+
+  ptype <- vec_type(df)
+  expect_is(ptype$y, "vctrs_unspecified")
```
Comment (hadley) [inline tests/testthat/test-unspecified.R]: I think this should be `expect_s3_class()`?

Comment (hadley) [reply tests/testthat/test-unspecified.R]: Or `expect_equal(ptype$y, unspecified())`?

File: tests/testthat/test-prototype.R [hunk 2]
```diff
@@ -39,7 +39,7 @@ test_that("unspecified prototypes created from under specified inputs", {
 
 test_that("unspecified prototypes created from data frame cols", {
```
[the remark this answers is not in the document]
Comment (lionel-) [reply tests/testthat/test-prototype.R]: What should the title be? It is now testing the same logic as the test above which has the same title structure.

[the remark this answers is not in the document]
Comment (lionel-) [reply tests/testthat/test-prototype.R]: The titles are now variants of: "finalised prototypes created from under specified inputs"

[the remark this answers is not in the document]
Comment (hadley) [reply tests/testthat/test-prototype.R]: LGTM

**Pull request merged**